CMakeLists.txt

```diff
@@ -104,6 +104,12 @@ endif()
 option(ZSTD_FROM_INTERNET "Download and use libzstd from the Internet" ${ZSTD_FROM_INTERNET_DEFAULT})
 find_package(zstd 1.1.2 REQUIRED)
 
+option(REDIS_STORAGE_BACKEND "Enable redis secondary storage" ON)
+if(REDIS_STORAGE_BACKEND)
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(HIREDIS REQUIRED hiredis>=0.13.3)
```

Comment:
CMake 3.10 is now required on master branch.

Reply:
The reported include directory in the pkg-config file is for using \<hiredis.h\>:

```
prefix=/home/anders/.linuxbrew/Cellar/hiredis/1.0.0
exec_prefix=${prefix}
libdir=/home/anders/.linuxbrew/Cellar/hiredis/1.0.0/lib
includedir=/home/anders/.linuxbrew/Cellar/hiredis/1.0.0/include/hiredis
```

So we need the prefix, or something like `../`, to use \<hiredis/hiredis.h\> ?

For the deb package we got "lucky" since /usr/include is already searched.

Reply:
To use `find_package`, we need to add a cmake file (for where it is missing).
It was easier to use the standard pkg-config, especially since it was broken...

But I made a change to add HIREDIS_FROM_INTERNET, and added one.
However, it didn't have any static library build target - only a shared library

Reply:
> The reported include directory in the pkg-config file is for using <hiredis.h>:

\*nod*

> [...] So we need the prefix, or something like `../`, to use <hiredis/hiredis.h> ?

Right, to use `#include <hiredis/hiredis.h>` we need `-I<pkg_config_prefix>/include` instead of `-I<pkg_config_includedir>`. But what I didn't understand was "adding the **homebrew** prefix" which sounded like it was something else. Did you mean "adding the **pkg-config** prefix"? If so I understand. :slightly_smiling_face:

> To use `find_package`, we need to add a cmake file (for where it is missing).

I realize that I was very unclear, sorry about that. What I should have written instead of "if using just `find_package`"  is something like "if using `find_package` and a `Findhiredis.cmake` file similar to how it's done for zstd". That is, in our own cmake code, use `find_library` and `find_path` to find the library and include directory in standard locations (or non-standard by e.g. setting `CMAKE_PREFIX_PATH`).

> It was easier to use the standard pkg-config, especially since it was broken...

I understand. If it's not too hard I would like to include Redis in the Windows builds as well, which is why I'm exploring alternatives here.

> But I made a change to add HIREDIS_FROM_INTERNET, and added one.

Now I don't follow again. I see no `HIREDIS_FROM_INTERNET`, so are you talking about some not yet published changes?

Reply:
Yeah, there were some merge conflicts. I will get it up soon, but a bit unsure about the shared lib.

Reply:
Added as 366c46461586222d1409170ad306a6901b83dcb9

Reply:
> But what I didn't understand was "adding the homebrew prefix" which sounded like it was something else. Did you mean "adding the pkg-config prefix"? If so I understand. slightly_smiling_face

I think I mixed them up a bit, since homebrew adds symlink into the individual installations from a global directory.
To confuse things further, this global directory is `/usr/local` on Mac - but I have it under `~/.linuxbrew` instead.

Reply:
Hmm, running from the package. Partly because I copied "lib" from zstd, partly because there is no "hiredis" directory.

-I/home/runner/work/ccache/ccache/build/hiredis-1.0.0/lib

build/hiredis-1.0.0/hiredis.h

So I'm not sure it is possible to use the build directory, without adding some workaround like symlinking hiredis to . ?

Reply:
If I understand correctly you made some changes after you wrote the above to make it work (copying headers to a hiredis directory)? Looks good to me.

Reply:
Since there are so many problems with it, I am going to leave HIREDIS_FROM_INTERNET and calling CMake.